spec/services/import_polygons_spec.rb

```diff
@@ -2,84 +2,151 @@
 
 RSpec.describe ImportPolygons do
   subject { described_class.new(location_type: location_type) }
-  let(:endpoint) { LOCATION_POLYGON_SETTINGS[location_type][:api] }
-  let(:response) { JSON.parse(file_fixture(file_name).read) }
+  let(:boundary_endpoint) { LOCATION_POLYGON_SETTINGS[location_type][:boundary_api] }
+  let(:boundary_response) { JSON.parse(file_fixture(file_name).read) }
 
   before do
-    allow(HTTParty).to receive(:get).with(endpoint).and_return(response)
-    subject.call
+    allow(HTTParty).to receive(:get).with(boundary_endpoint).and_return(boundary_response)
   end
 
   describe "#call" do
-    context "when location type is regions" do
-      let(:location_type) { :regions }
-      let(:points) { [55.8110853660943, -2.0343575091738, 55.7647624900862, -1.9841097397706] }
-      let(:file_name) { "ons_regions.json" }
-
-      it "imports North East region" do
-        expect(LocationPolygon.regions.first.name).to eq("north east")
+    context "when stubbing #get_buffers" do
+      before do
+        allow(subject).to receive(:get_buffers).with(boundary_points).and_return({ "5" => %w[lat lon], "10" => %w[lat lon] })
+        subject.call
       end
 
-      it "does not import non-existent region" do
-        expect(LocationPolygon.regions.count).to eq(1)
-      end
+      context "when location type is regions" do
+        let(:location_type) { :regions }
+        let(:boundary_points) { [55.8110853660943, -2.0343575091738, 55.7647624900862, -1.9841097397706] }
+        let(:file_name) { "ons_regions.json" }
 
-      it "imports Nort East boundaries" do
-        expect(LocationPolygon.regions.first.boundary).to eq(points)
-      end
-    end
+        it "imports North East region" do
+          expect(LocationPolygon.regions.first.name).to eq("north east")
+        end
 
-    context "when location type is counties" do
-      let(:location_type) { :counties }
-      let(:points) { [52.0652732493605, 0.7031775640868, 52.0486871974891, 0.7162633765872] }
-      let(:file_name) { "ons_counties.json" }
+        it "does not import non-existent region" do
+          expect(LocationPolygon.regions.count).to eq(1)
+        end
 
-      it "imports Essex county" do
-        expect(LocationPolygon.counties.first.name).to eq("essex")
+        it "imports Nort East boundaries" do
+          expect(LocationPolygon.regions.first.boundary).to eq(boundary_points)
+        end
       end
 
-      it "does not import non-existent county" do
-        expect(LocationPolygon.counties.count).to eq(1)
-      end
+      context "when location type is counties" do
+        let(:location_type) { :counties }
+        let(:boundary_points) { [52.0652732493605, 0.7031775640868, 52.0486871974891, 0.7162633765872] }
```

Comment:
If you changed this, you could probably combine the ons_counties.json fixture and the new fixture into one fixture.